DDF-1506: Enable CRL Checking for Web Contexts

garrettfreibott opened the pull request:
@pklinef 
@stustison 
@shaundmorris 
@lessarderic 
@djblue 
@andrewkfiedler is the hero

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf/252)
<!-- Reviewable:end -->

brendan-hofmann commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3297 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bQa49u1dLvEwCi_2k)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3297)):</sup>
'CRL Uncommenting...'
Is that a typo? Seems like an odd way to start a sentence.
<!-- Reviewable comment -K0bQa49u1dLvEwCi_2l:1068337651 -->

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3317 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bQodFZxt3vzkh49Tz)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3317)):</sup>
typo: revoced/revoked
<!-- Reviewable comment -K0bQodGquuxbIhNPwou:317638550 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 49 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bR7hEdKnnuaHOllrN)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L49)):</sup>
Why not use the existing constant in wss4j's Merlin.java?
<!-- Reviewable comment -K0bR7hEdKnnuaHOllrO:-1195493137 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 129 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bRNbDb17xXN62l455)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L129)):</sup>
"to attached"
Shouldn't that be "to be attached"?
<!-- Reviewable comment -K0bRNbE_RatdP39b9BH:1505341515 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 257 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bS4Iys1Hn6GrP5to0)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L257)):</sup>
If this is only for unit tests it should be package-private.
<!-- Reviewable comment -K0bS4Iys1Hn6GrP5to1:1208651093 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 278 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bSDdJFjrEp82nxfn7)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L278)):</sup>
Again, package-private.
<!-- Reviewable comment -K0bSDdKpbtaSPvOnZlq:-877269311 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

harrison-tarr commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

shaundmorris commented:
Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3321 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bRfQK8GpK-Iiw5ms1)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3321)):</sup>
Typo "is not required IF the CRL is edited"
<!-- Reviewable comment -K0bRfQLvUjvute7csc-:-195437329 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

AzGoalie commented:
Reviewed 8 of 9 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions, all commit checks successful.

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3321 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bQxE0Q9rhOGo2sIqI)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3321)):</sup>
Typo: 'is not required is' should be 'is not require if'
<!-- Reviewable comment -K0bQxE0Q9rhOGo2sIqJ:-693731638 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 165 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bUjeCwFT5OJ5L857o)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L165)):</sup>
Should still use assertThat here. Example assertThat(crlRelativePath, is(nullValue()));
<!-- Reviewable comment -K0bUjeCwFT5OJ5L857p:-1260208651 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Review status: 6 of 9 files reviewed at latest revision, 9 unresolved discussions, some commit checks pending.

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3297 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bQa49u1dLvEwCi_2k)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3297)):</sup>
Yup. just needed to remove "CRL".
<!-- Reviewable comment -K0flFMdQR9yJbrhLCen:-308204554 -->

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3317 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bQodFZxt3vzkh49Tz)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3317)):</sup>
These typos will be the end of me!
<!-- Reviewable comment -K0flhcWAozxJ1VWxNdR:547003478 -->

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3321 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bQxE0Q9rhOGo2sIqI)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3321)):</sup>
Got it!
<!-- Reviewable comment -K0flr8aQrLTDgNHvHhp:-271887875 -->

---

<sup>**[distribution/docs/src/main/resources/ManContents.adoc, line 3321 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bRfQK8GpK-Iiw5ms1)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/distribution/docs/src/main/resources/ManContents.adoc#L3321)):</sup>
Got it!
<!-- Reviewable comment -K0fmCKNE2oj-1PTifhq:-271887875 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 49 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bR7hEdKnnuaHOllrN)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L49)):</sup>
Awesome, I didn't know they had constants for the encryption.properties file in there. I'll set this final to Merlin.OLD_PREFIX + Merlin.X509_CRL_FILE.
<!-- Reviewable comment -K0fnmRrgFeAT5uLqDP6:280030132 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 129 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bRNbDb17xXN62l455)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L129)):</sup>
Yes, seem to have accidentally chose a weird middle ground between "to be attached" and "to attach".
<!-- Reviewable comment -K0fqfQFOoa5gkoL7IQi:-709728441 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 257 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bS4Iys1Hn6GrP5to0)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L257)):</sup>
Noted, I fixed the unit test methods.
<!-- Reviewable comment -K0fqzj1u9vRfhWjb6ii:-989181247 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 278 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bSDdJFjrEp82nxfn7)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L278)):</sup>
Got it.
<!-- Reviewable comment -K0fr25aOXP3oUUlmKq-:-271887862 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 165 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bUjeCwFT5OJ5L857o)** ([raw file](https://github.com/codice/ddf/blob/f7a357a7fdfaf625ef4c4525e7c6de4bd256aa14/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L165)):</sup>
Can do, I'll fix the assertNulls.
<!-- Reviewable comment -K0fra8zbflbXOHduKkB:334560624 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

stustison commented on platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java:
```diff
@@ -68,9 +90,63 @@ public HandlerResult getNormalizedToken(ServletRequest request, ServletResponse
+
+        // Somehow the httpResponse was null, return no action and try to process with other handlers
+        if (httpResponse == null) {
+            LOGGER.error("Somehow the httpResponse was null");
```
Lets give a better error message than this. Maybe something like "HTTP Response was null for request: <path they are trying to hit>"

stustison commented on platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java:
```diff
@@ -85,10 +161,122 @@ public HandlerResult handleError(ServletRequest servletRequest, ServletResponse
+     */
+    private CRL createCRL(String location)
+            throws FileNotFoundException, CertificateException, CRLException {
+        FileInputStream fis = new FileInputStream(new File(location));
```
Need to close this stream. do a try with resources here.

shaundmorris commented:
Reviewed 1 of 3 files at r2.
Review status: 7 of 9 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 248 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bF7kRZiVx1enXo5IE-r2-248)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L248)):</sup>
Good catch on the InputStream. We probably should do the same in the unit tests as well.
<!-- Reviewable comment -K0gLq4vwUjE3z9E7J2m:705448684 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

andrewkfiedler commented:
:+1: built and tested

lessarderic commented:
Review status: 7 of 9 files reviewed at latest revision, 15 unresolved discussions, all commit checks successful.

---

<sup>**[platform/security/handler/security-handler-pki/pom.xml, line 100 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h1coq8JapfWAzrrQf)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/pom.xml#L100)):</sup>
:(
<!-- Reviewable comment -K0g8qihO5reZ4CKzFS4:5172979 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 52 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h1jf1A-Ga3YNSG-0C)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L52)):</sup>
Can you make this private if it isn't used in sub-classes?
<!-- Reviewable comment -K0h1jf1A-Ga3YNSG-0D:-241134782 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 112 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h3GRh6YWEXIbFcHQH)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L112)):</sup>
Instead of setting `crl` null and check for null everywhere else we need to use it, we could create a separate `CrlChecker` class that contains the `checkAgainstCrl()` method and any other method related to CRL and have it be a no-op object if the CRL isn't set. If you feel it's too late to make that change you can always open an improvement ticket for this.
<!-- Reviewable comment -K0h3GRh6YWEXIbFcHQI:-1823689075 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 135 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h53lr3NERwSIysJru)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L135)):</sup>
Can this be private or protected instead of public? Should also camel-case CRL in the name.
<!-- Reviewable comment -K0h53lr3NERwSIysJrv:1868755574 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 170 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h5EuXWHBamETA_qID)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L170)):</sup>
Same comment as above.
<!-- Reviewable comment -K0h5EuXWHBamETA_qIE:886361481 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 176 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h6iUgqR1zp_KIvwbe)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L176)):</sup>
Should we build the message once here like we did in the `else` statement?
<!-- Reviewable comment -K0h6iUhmFoTxWd-91MM:1147477502 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 211 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h6xrxK1AaqYAPw5-t)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L211)):</sup>
Do we expect this method to be called from the outside? Does it have to be public?
It's also kind of odd that only this method is `synchronized`.  We should synchronize on reads too if we need to synchronize here.
<!-- Reviewable comment -K0h6xrxK1AaqYAPw5-u:1804994605 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 222 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h7biBnNZiYCn_RZ_b)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L222)):</sup>
Do we really need the `isEnabled` flag? Can't we just use the `crl` object reference? If we extract a `CrlChecker` class, the flag could go away and that logic wouldn't be needed anymore.
<!-- Reviewable comment -K0h7biBnNZiYCn_RZ_c:1533915521 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 246 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h6u2rJ3_0_CyQtxAY)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L246)):</sup>
`createCrl()`
<!-- Reviewable comment -K0h6u2rJ3_0_CyQtxAZ:1277049881 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

roelens8 commented on platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java:
```diff
@@ -68,9 +90,63 @@ public HandlerResult getNormalizedToken(ServletRequest request, ServletResponse
+    }
+
+    /**
+     * Checks the certificates agains the CRL. If it is in the CRL, send a 401 error and return a HandlerResult with
```
against*

mrmateo commented on platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java:
```diff
@@ -0,0 +1,196 @@
+     */
+    private CRL createCRL(String location)
+            throws FileNotFoundException, CertificateException, CRLException {
+        try (FileInputStream fis = new FileInputStream(new File(location))) {
```
Would be useful if you could abstract this so that the CRL file could also be a URL.

mrmateo commented:
Looks like there's a good amount of code duplication between this and the CRLInterceptor (https://github.com/codice/ddf/blob/master/platform/security/sts/security-sts-crlinterceptor/src/main/java/org/codice/ddf/security/sts/crl/CRLInterceptor.java). Is there a possibility of merging these to remove the duplicate code? Maybe just have the interceptor call this code underneath the scenes?

pklinef commented on platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java:
```diff
@@ -0,0 +1,196 @@
+
+        crl = null;
+        try {
+            crl = createCRL(location);
```
Why does createCRL handle the IOException but not these other exceptions?

pklinef commented on platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java:
```diff
@@ -0,0 +1,196 @@
+                httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+                httpResponse.flushBuffer();
+            } catch (IOException e) {
+                LOGGER.debug("Unable to send 401 response to client.");
```
Shouldn't we throw a servlet exception if we are not able to write a 401 for a user trying to use a revoked certificate?  The web sso filter assumes the handler did everything needed when the status is redirected.

garrettfreibott commented:
I will create a ticket to address modifying the CRLInterceptor to use this class due to time restraints, and I'll get to work on that soon. The code in the CrlChecker class has been modified enough at this point not to be easily integrated into the CRLInterceptor,  so some deliberation will need to be made to rethink the design of the CRLInterceptor and whether or not it should behave identically to the AbstractPKIHandler's implementation. The CrlInterceptor class also just needs some reworking as it is.
<!-- Reviewable comment -K0n-6SAz7gPKd9uavSf:681694541 -->

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Review status: 5 of 12 files reviewed at latest revision, 19 unresolved discussions, some commit checks pending.

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 102 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bF7kRZiVx1enXo5IE-r2-102)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L102)):</sup>
Got it.
<!-- Reviewable comment -K0h3f_cl_eXxVH5Bn4t:-271887862 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java, line 248 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0bF7kRZiVx1enXo5IE-r2-248)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/AbstractPKIHandler.java#L248)):</sup>
I'll add the try w/ resources here, but I don't have any file streams in the unit tests that need to be closed that I can see. The file reads use PropertiesLoader that takes care of closing it's streams. I also made this change to the same method in the CRLInterceptor.
<!-- Reviewable comment -K0h0oflAfzqarNiM8ZG:-1458509386 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 148 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0hweqzvAxDN_BdeAIg-r4-148)** ([raw file](https://github.com/codice/ddf/blob/6ee28ff9a034a420f320bb113d2c1b32de69dcff/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L148)):</sup>
It was a bit of weirdness carried over from the CrlInterceptor. Should look a bit better now.
<!-- Reviewable comment -K0n0ykAiaPXnuev2OiK:-1465842835 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 177 \[r4\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0hweqzvAxDN_BdeAIg-r4-177)** ([raw file](https://github.com/codice/ddf/blob/6ee28ff9a034a420f320bb113d2c1b32de69dcff/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L177)):</sup>
I'll create a ticket for this as well, as it's a bit outside of the scope of the original ticket.
<!-- Reviewable comment -K0n18sQbTAB8LF1rWXm:-333977470 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 2 of 7 files at r5, 1 of 3 files at r6.
Review status: 8 of 12 files reviewed at latest revision, 28 unresolved discussions, all commit checks successful.

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 60 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x8eLJ2TvEGBHc4xk-)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L60)):</sup>
Info seems a little high. Not sure we want to see this every time in the logs.
<!-- Reviewable comment -K0x8eLKPwSwRMKnh7G-:1651366690 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 86 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x9XfhELaufiOFX4tR)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L86)):</sup>
Should the comment say `Cert is *not*  present in the CRL list` ?
<!-- Reviewable comment -K0x9XfhELaufiOFX4tS:187560141 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 93 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x9hEMM9aE_KLfFphB)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L93)):</sup>
Should log an error here.
<!-- Reviewable comment -K0x9hEMM9aE_KLfFphC:-1879641833 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 105 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x99dyEHSVteUxhn2B)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L105)):</sup>
Should be `passesCrl` (camel-cased). And since this is a private method, you can delete the javadoc comments. Otherwise I'll ask you to add descriptions to parameter and return value :-)
<!-- Reviewable comment -K0x99dyEHSVteUxhn2C:-1286529451 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 134 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xAspzZHPJBPhXJ5vJ)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L134)):</sup>
Should we log something to the security log file?
<!-- Reviewable comment -K0xAspzZHPJBPhXJ5vK:1597142159 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 148 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xAUA9o5t8io6OfIvL)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L148)):</sup>
Should be `createCrl` (camel-cased). And same comment regarding the javadocs on private methods.
<!-- Reviewable comment -K0xAUA9o5t8io6OfIvM:-315096155 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 156 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xB29PL3LbB5vLOC4r)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L156)):</sup>
I would add the file location to the log and up-case CRL.
<!-- Reviewable comment -K0xB29PL3LbB5vLOC4s:807167724 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 167 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xBBz7yuaTokkYnhUl)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L167)):</sup>
This could be package private instead.
<!-- Reviewable comment -K0xBBz7yuaTokkYnhUm:-237835644 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/WssPKIHandler.java, line 33 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xBUZYbuVogydbdLXC)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/WssPKIHandler.java#L33)):</sup>
This constructor shouldn't be needed since the super class has a default constructor.
<!-- Reviewable comment -K0xBUZYbuVogydbdLXD:353912291 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 40 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xE1XGwHz677Jq3eXw)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L40)):</sup>
This test class is actually testing both, `PKIHandler`  and `CrlChecker`. If you have time, it might be worth splitting it up to match the classes we have. Would make the tests easier to understand and maintain.
<!-- Reviewable comment -K0xE1XGwHz677Jq3eXx:-1722100468 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 47 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xDrT_Cr93NvXNHIbl)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L47)):</sup>
Should camel-case `CRL` and `PKI` in the variable, fields and method names.
<!-- Reviewable comment -K0xDrTaQzTmL1r9Vl5j:204465194 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 51 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xBxtHtHYFqadY5l70)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L51)):</sup>
Any reason we're not using a mock for this?
<!-- Reviewable comment -K0xBxtHtHYFqadY5l71:-1950034605 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 77 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xEMpqFf7-RZ9JkVYX)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L77)):</sup>
Can we assert something more specific? If we were to mock the `CrlChecker`, we could simply assert that `result`  is the same instance as the one returned by the `CrlChecker`.
<!-- Reviewable comment -K0xEMprNjBJRB4NHcrJ:567677474 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

lessarderic commented:
Reviewed 3 of 7 files at r5, 1 of 3 files at r6.
Review status: all files reviewed at latest revision, 28 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Review status: 7 of 12 files reviewed at latest revision, 27 unresolved discussions, some commit checks pending.

---

<sup>**[platform/security/handler/security-handler-pki/pom.xml, line 100 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0h1coq8JapfWAzrrQf)** ([raw file](https://github.com/codice/ddf/blob/732b5b3a038d6204f0771210ec987aff0b01105b/platform/security/handler/security-handler-pki/pom.xml#L100)):</sup>
It's better now! :)
<!-- Reviewable comment -K0n1HE-ntFzopB_Vj62:-980861032 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 60 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x8eLJ2TvEGBHc4xk-)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L60)):</sup>
I changed it to logTrace.
<!-- Reviewable comment -K1-N0dZYmW6AkRPcLqd:-1000420427 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 86 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x9XfhELaufiOFX4tR)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L86)):</sup>
It hits this point, it means that the cert is in the CRL so it should not be allowed past. I changed the wording to "cert is present and listed as revoked in the CRL" be a bit more clear.
<!-- Reviewable comment -K1-N4UWEge55ez81J7v:34599594 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 93 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x9hEMM9aE_KLfFphB)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L93)):</sup>
Logged.
<!-- Reviewable comment -K1-OIQGTBl5ZIJrOtcQ:-139231705 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 105 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0x99dyEHSVteUxhn2B)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L105)):</sup>
I like having javadocs on private methods in case someone later needs to work in this class. I added descriptions to the parameter and return.
<!-- Reviewable comment -K1-OK_VasS2lkw9mWs_:783269268 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 134 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xAspzZHPJBPhXJ5vJ)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L134)):</sup>
Logged
<!-- Reviewable comment -K1-PHnQpyFTALDeibce:1935171303 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 148 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xAUA9o5t8io6OfIvL)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L148)):</sup>
Camel'd and description'd.
<!-- Reviewable comment -K1-PO5w37Kz-IpJY0Qh:1559316592 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 156 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xB29PL3LbB5vLOC4r)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L156)):</sup>
Location added.
<!-- Reviewable comment -K1-QFukNl2SMgtwaubB:8469044 -->

---

<sup>**[platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java, line 167 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xBBz7yuaTokkYnhUl)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java#L167)):</sup>
Got it!
<!-- Reviewable comment -K1-QLV-1EFdhRbL1P2A:-271887875 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Review status: 7 of 12 files reviewed at latest revision, 20 unresolved discussions, some commit checks pending.

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 40 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xE1XGwHz677Jq3eXw)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L40)):</sup>
I can make a ticket for it and assign myself to it. It's getting a bit down to the wire as far as time is concerned.
<!-- Reviewable comment -K1-Y2yzwwik2Dg1JbDd:102136362 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Review status: 7 of 12 files reviewed at latest revision, 19 unresolved discussions, some commit checks pending.

---

<sup>**[platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java, line 77 \[r6\]](https://reviewable.io:443/reviews/codice/ddf/252#-K0xEMpqFf7-RZ9JkVYX)** ([raw file](https://github.com/codice/ddf/blob/d5e930e080f03eb9c43abaa62d954ae7ea46212d/platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/PKIHandlerTest.java#L77)):</sup>
I'll add this task and the mock task above to the refactoring ticket, as well.
<!-- Reviewable comment -K1-ZB5_LF4akdpQkEqF:-193842479 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/252)
<!-- Sent from Reviewable.io -->

garrettfreibott commented:
Here are the tickets created from some of the feedback, they'll be worked right after this one:
https://codice.atlassian.net/browse/DDF-1573
https://codice.atlassian.net/browse/DDF-1574

shaundmorris commented:
Verified build on bamboo. did a final manual build (minus documentation). Verified comments were addressed on this PR or scheduled to be addressed on other ticket(s).
